### PR TITLE
Add BraidTestProvider component

### PR DESCRIPTION
**Pull request by markdalgleish**
See changeset for details.

### CODE REVIEW

File: lib/components/BraidTestProvider/BraidTestProvider.docs.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import { seekAnz } from '../../themes';
+import { ComponentDocs } from '../../../site/src/types';
+import { Stack, Text } from '..';
+
+const docs: ComponentDocs = {
+  category: 'Logic',
+  description: (
+    <Stack space="large">
+      <Text>Alternative to `BraidProvider` for unit test environments.</Text>
```
Comment (mattsjones) [inline lib/components/BraidTestProvider/BraidTestProvider.docs.tsx]: Could we add a strong warning to not import this in your actual App? Just to be sure.

Comment (markdalgleish) [reply lib/components/BraidTestProvider/BraidTestProvider.docs.tsx]: Maybe even throw an error?

Comment (mattsjones) [reply lib/components/BraidTestProvider/BraidTestProvider.docs.tsx]: How would we detect that?

Comment (markdalgleish) [reply lib/components/BraidTestProvider/BraidTestProvider.docs.tsx]: `NODE_ENV` of development or production?

Comment (mattsjones) [reply lib/components/BraidTestProvider/BraidTestProvider.docs.tsx]: Worried that some teams will have that set for some reason in there test setup.

Comment (markdalgleish) [reply lib/components/BraidTestProvider/BraidTestProvider.docs.tsx]: Yeah, fair point. I'll add it to the docs.

**Pull request merged**